crates/katana-core/src/starknet/mod.rs

```diff
@@ -349,7 +353,18 @@ impl StarknetWrapper {
 
     fn update_block_context(&mut self) {
         self.block_context.block_number = self.block_context.block_number.next();
-        self.block_context.block_timestamp = BlockTimestamp(get_current_timestamp().as_secs());
+        if self.block_context_generator.next_block_start_time == 0 {
+            self.block_context.block_timestamp = BlockTimestamp(
+                get_current_timestamp().as_secs()
+                    + self.block_context_generator.block_timestamp_offset,
+            );
+        } else {
+            let block_timestamp = self.block_context_generator.next_block_start_time;
+            self.block_context_generator.block_timestamp_offset =
+                block_timestamp - get_current_timestamp().as_secs();
```

Comment:
wouldn't this panic if user set the next timestamp smaller than the current one?